lib/imapoptions

```diff
@@ -2422,7 +2422,8 @@ product version in the capabilities
    modified UTF7 encoding should be used. */
 
 { "sieve_sasl_send_unsolicited_capability", 0, SWITCH, "2.3.17" }
```

Comment:
You should also update the "last_modified" and "deprecated_since" field of this config item, which you do by changing the existing last_modified version (currently "2.3.17") and adding the deprecated_since version after it.  Since you don't know what version this will be released in, you use the string "UNRELEASED" for both fields, and the release manager (me) will sort it out when the time comes.

Which is to say, you should change this line to something like this:

`{ "sieve_sasl_send_unsolicited_capability", 0, SWITCH, "UNRELEASED", "UNRELEASED" }`

This will allow `cyr_info` to recognise that the option has changed, and that it's deprecated, and will enable the deprecated config warning if it's in imapd.conf at startup. :)

Formally deprecating it like this will cause a fatal assertion if any part of Cyrus tries to use the option, so you'll also need to update timsieved/parser.c to no longer try to read it, probably by removing these lines around line 889:

```
  if (sasl_ssf &&
      config_getswitch(IMAPOPT_SIEVE_SASL_SEND_UNSOLICITED_CAPABILITY)) {
      capabilities(sieved_out, sieved_saslconn, starttls_done, authenticated,
                   sasl_ssf);
      prot_flush(sieved_out);
  }
```

Nothing else seems to use the option at all

Reply:
> Formally deprecating it like this will cause a fatal assertion if any part of Cyrus tries to use the option, 

What is the difference between “Crashing, because undefined option was used” and “Crashing, because deprecated option was used”?

Reply:
We don't have undefined options, because we deprecate old ones, not remove them entirely.  If we tried to use an option that wasn't defined, it wouldn't even compile, because IMAPOPT_WHATEVER wouldn't exist in the enum.

Asserting if Cyrus still tries to use them after they're deprecated tells us (devs) that we screwed up, and tried to use something that we told sysadmins doesn't exist anymore.  You can see the full range of deprecated behaviours in the test_deprecated_* tests in cunit/libconfig.testc if you're curious.

If you really really need to, you can read an option from imapd.conf that isn't defined in imapoptions using the `config_getoverflowstring()` API, which is how things like the channel prefix on sync options are handled.  But this API doesn't provide the SWITCH/INT/DURATION/etc parsing or type checking, it just gives you the string value from the config file and you have to parse it yourself.  So if we really wanted to deprecate the option but still honour it if it were set, we could convert the existing  `config_getswitch(IMAPOPT_SIEVE_SASL_SEND_UNSOLICITED_CAPABILITY)` check to read it as an overflow string and figure out its true/false value manually, but that seems like a lot of work for something that shouldn't even exist?